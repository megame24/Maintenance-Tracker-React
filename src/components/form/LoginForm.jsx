import React from 'react';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router-dom';
import Loading from '../Loading';
import ErrorMessage from '../ErrorMessage';

const LoginForm = (props) => {
  const {
    handleSubmit, formData, handleChange,
    isLoading, errors,
  } = props;
  return (
    <section id="login">
      <Loading isLoading={isLoading} />
      <div className="container">
        <ErrorMessage errors={errors} />
        <div className="form">
          <div className="form-icon-container">
            <div className="center form-icon blue">
              <i className="icon ion-person" />
            </div>
          </div>
          <h1 className="center">Create Account</h1>
          <form id="LoginForm" onSubmit={handleSubmit}>
            <input
              onChange={handleChange}
              value={formData.usernameOrEmail}
              className="form-input"
              type="text"
              name="usernameOrEmail"
              id="email"
              placeholder="Email or Username"
              required
            />
            <input
              onChange={handleChange}
              value={formData.password}
              className="form-input"
              type="password"
              name="password"
              id="password"
              placeholder="Password"
              required
            />
            <input
              className="form-input btn btn-primary"
              type="submit"
              id="submit-btn"
              value="Login"
            />
          </form>
          <div className="form-footer center">
            <p>
              Not a member? &nbsp;
              <Link to="/signup" className="blue">Create Account</Link>
            </p>
          </div>
        </div>
      </div>
    </section>
  );
};

LoginForm.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  formData: PropTypes.shape({
    password: PropTypes.string.isRequired
  }).isRequired,
  isLoading: PropTypes.bool.isRequired,
  errors: PropTypes.shape({
    statusCode: PropTypes.number
  }),
};

LoginForm.defaultProps = {
  errors: {},
};

export default LoginForm;
