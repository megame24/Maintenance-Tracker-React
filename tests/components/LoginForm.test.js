import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from '../../src/components/form/LoginForm';

describe('testing LoginForm component', () => {
  const props = {
    isLoading: false,
    handleChange: () => {},
    handleSubmit: () => {},
    formData: {
      usernameOrEmail: '',
      password: '',
    }
  }
  it('should render as expected when there is no error', () => {
    const tree = shallow(<LoginForm {...props} />)
    expect(tree).toMatchSnapshot();
  });
  it('should render as expected when there is an error', () => {
    props.errors = {
      message: 'error'
    };
    const tree = shallow(<LoginForm {...props} />)
    expect(tree).toMatchSnapshot();
  });
});

