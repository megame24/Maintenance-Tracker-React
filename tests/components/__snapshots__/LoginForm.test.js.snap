// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing LoginForm component should render as expected when there is an error 1`] = `
<section
  id="login"
>
  <Loading
    isLoading={false}
  />
  <div
    className="container"
  >
    <ErrorMessage
      errors={
        Object {
          "message": "error",
        }
      }
    />
    <div
      className="form"
    >
      <div
        className="form-icon-container"
      >
        <div
          className="center form-icon blue"
        >
          <i
            className="icon ion-person"
          />
        </div>
      </div>
      <h1
        className="center"
      >
        Create Account
      </h1>
      <form
        id="LoginForm"
        onSubmit={[Function]}
      >
        <input
          className="form-input"
          id="email"
          name="usernameOrEmail"
          onChange={[Function]}
          placeholder="Email or Username"
          required={true}
          type="text"
          value=""
        />
        <input
          className="form-input"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <input
          className="form-input btn btn-primary"
          id="submit-btn"
          type="submit"
          value="Login"
        />
      </form>
      <div
        className="form-footer center"
      >
        <p>
          Not a member?  
          <Link
            className="blue"
            replace={false}
            to="/signup"
          >
            Create Account
          </Link>
        </p>
      </div>
    </div>
  </div>
</section>
`;

exports[`testing LoginForm component should render as expected when there is no error 1`] = `
<section
  id="login"
>
  <Loading
    isLoading={false}
  />
  <div
    className="container"
  >
    <ErrorMessage
      errors={Object {}}
    />
    <div
      className="form"
    >
      <div
        className="form-icon-container"
      >
        <div
          className="center form-icon blue"
        >
          <i
            className="icon ion-person"
          />
        </div>
      </div>
      <h1
        className="center"
      >
        Create Account
      </h1>
      <form
        id="LoginForm"
        onSubmit={[Function]}
      >
        <input
          className="form-input"
          id="email"
          name="usernameOrEmail"
          onChange={[Function]}
          placeholder="Email or Username"
          required={true}
          type="text"
          value=""
        />
        <input
          className="form-input"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <input
          className="form-input btn btn-primary"
          id="submit-btn"
          type="submit"
          value="Login"
        />
      </form>
      <div
        className="form-footer center"
      >
        <p>
          Not a member?  
          <Link
            className="blue"
            replace={false}
            to="/signup"
          >
            Create Account
          </Link>
        </p>
      </div>
    </div>
  </div>
</section>
`;
